projects/clr/rocclr/device/rocm/rocdevice.cpp

```diff
@@ -2365,11 +2365,13 @@ bool Device::virtualFree(void* addr) {
   return true;
 }
 
-bool Device::SetMemAccess(void* va_addr, size_t va_size, VmmAccess access_flags) {
+bool Device::SetMemAccess(void* va_addr, size_t va_size, VmmAccess access_flags,
+                          VmmLocationType access_location) {
   hsa_status_t hsa_status = HSA_STATUS_SUCCESS;
   hsa_amd_memory_access_desc_t desc;
   desc.permissions = static_cast<hsa_access_permission_t>(access_flags);
-  desc.agent_handle = getBackendDevice();
+  desc.agent_handle =
```

Comment:
What happens if the memory is not allocated using hipHostMalloc or on the host side? Should we have a check at CLR level? If we rely on ROCr to respond the right error code, did we check to make sure we return the right error code similar to CUDA?